src/components/project-input.ts

```diff
@@ -1,74 +1,129 @@
 //Project input class
-///<reference path="base-components.ts"/>
-///<reference path="../util/validation.ts"/>
-///<reference path="../decorators/autobind.ts"/>
-///<reference path="../state/project-state.ts"/>
-
-namespace App {
- export class ProjectInput extends Component<HTMLDivElement, HTMLFormElement> {
-    titleInputElement: HTMLInputElement;
-    descriptionInputElement: HTMLInputElement;
-    peopleInputElement: HTMLInputElement;
-    constructor() {
-      super("project-input", "app", true, "user-input");
-      this.titleInputElement = this.element.querySelector(
-        "#title"
-      )! as HTMLInputElement;
-      this.descriptionInputElement = this.element.querySelector(
-        "#description"
-      )! as HTMLInputElement;
-      this.peopleInputElement = this.element.querySelector(
-        "#people"
-      )! as HTMLInputElement;
-      this.configure();
-    }
-    configure() {
-      this.element.addEventListener("submit", this.submitHandler);
+///<
+import { Component } from "./base-components";
+import { Validatable, validate } from "../util/validation";
+import { projectState } from "../state/project-state";
+//ProjectInput:class which takes care of the user input with validation and event listener for submitting
+export class ProjectInput extends Component<HTMLDivElement, HTMLFormElement> {
+  titleInputElement: HTMLInputElement;
+  descriptionInputElement: HTMLInputElement;
+  peopleInputElement: HTMLInputElement;
+  constructor() {
+    //super(the id of the template we gonna use,where we gonna store the values(id of the host),
+    //if we want to add at start(true) else at the
+    //end,the id we want to give to the element if we want)
+    super("project-input", "app", true, "user-input");
+    this.titleInputElement = this.element.querySelector(
+      "#title"
+    )! as HTMLInputElement;
+    this.descriptionInputElement = this.element.querySelector(
+      "#description"
+    )! as HTMLInputElement;
+    this.peopleInputElement = this.element.querySelector(
+      "#people"
+    )! as HTMLInputElement;
+    this.configure();
+  }
+  configure() {
+    this.element.addEventListener("submit", this.submitHandler);
+  }
+  renderContent(): void {}
+  //on submit
+  private submitHandler=(event: Event)=> {
+    event.preventDefault();
+    const userInput = this.gatherUserInput();
+    //if user input is correct we receive the array
+    if (Array.isArray(userInput)) {
+      const [title, description, people] = userInput;
```

Comment:
why are you destructing the userInput just to pass it as parameters on the next line when you can just have one line like 
projectState.addProject(userInput[0], userInput[1], userInput[2]) .
Its not wrong , and it is a bit more descriptive which is fine, but if there is no manipulation of the fields in any way you can just pass them from the array.